consensus/src/state_computer.rs

```diff
@@ -33,14 +36,16 @@ pub struct ExecutionProxy {
     mempool_notifier: Arc<dyn TxnManager>,
     state_sync_notifier: Arc<dyn ConsensusNotificationSender>,
     async_state_sync_notifier: channel::Sender<NotificationType>,
+    // TODO: is a mutex needed here? if not, what?
```

Comment:
unfortunately this object is shared across threads, we need Mutex for mutability. I'm still thinking if we can avoid this.

one idea comes to my mind is to change this to a map<index, sig> https://github.com/aptos-labs/aptos-core/blob/b1601d23fb157fee4c4932cd093d3809be8094a4/types/src/ledger_info.rs#L198

but we can do that in a followup PR

Reply:
let's remove the TODO